File: drs/src/test/com/db4o/drs/test/UuidConversionTestCase.java
/* Copyright (C) 2004 - 2010  Versant Inc.  http://www.db4o.com */

package com.db4o.drs.test;

import com.db4o.drs.foundation.*;
import com.db4o.drs.inside.*;
import com.db4o.drs.test.data.*;

import db4ounit.*;

public class UuidConversionTestCase extends DrsTestCase{
	
	
	public void test() {
		SPCChild child = storeInA();
		replicate();
		ReplicationReference ref = a().provider().produceReference(child);
		b().provider().clearAllReferences();
		DrsUUID expectedUuid = ref.uuid();
		ReplicationReference referenceByUUID = b().provider().produceReferenceByUUID(expectedUuid, null);
		Assert.isNotNull(referenceByUUID);
		DrsUUID actualUuid = referenceByUUID.uuid();
		Assert.areEqual(expectedUuid.getLongPart(), actualUuid.getLongPart());
	}

	private SPCChild storeInA() {
		String name = "c1";
		SPCChild child = createChildObject(name);
		
		a().provider().storeNew(child);
		a().provider().commit();
		return child;
	}
	
	private void replicate() {
		replicateAll(a().provider(), b().provider());
	}
	
	protected SPCChild createChildObject(String name) {
		return new SPCChild(name);
	}



}

File: db4o.net/drs/Db4objects.Drs.Tests/Db4objects.Drs.Tests/UuidConversionTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4objects.Drs.Foundation;
using Db4objects.Drs.Inside;
using Db4objects.Drs.Tests;
using Db4objects.Drs.Tests.Data;

namespace Db4objects.Drs.Tests
{
	public class UuidConversionTestCase : DrsTestCase
	{
		public virtual void Test()
		{
			SPCChild child = StoreInA();
			Replicate();
			IReplicationReference @ref = A().Provider().ProduceReference(child);
			B().Provider().ClearAllReferences();
			IDrsUUID expectedUuid = @ref.Uuid();
			IReplicationReference referenceByUUID = B().Provider().ProduceReferenceByUUID(expectedUuid
				, null);
			Assert.IsNotNull(referenceByUUID);
			IDrsUUID actualUuid = referenceByUUID.Uuid();
			Assert.AreEqual(expectedUuid.GetLongPart(), actualUuid.GetLongPart());
		}

		private SPCChild StoreInA()
		{
			string name = "c1";
			SPCChild child = CreateChildObject(name);
			A().Provider().StoreNew(child);
			A().Provider().Commit();
			return child;
		}

		private void Replicate()
		{
			ReplicateAll(A().Provider(), B().Provider());
		}

		protected virtual SPCChild CreateChildObject(string name)
		{
			return new SPCChild(name);
		}
	}
}
